src/functional/ffront/past_to_itir.py

```diff
@@ -205,6 +217,39 @@ def _construct_itir_domain_arg(
 
         return itir.FunCall(fun=itir.SymRef(id=domain_builtin), args=domain_args)
 
+    def _construct_itir_initialized_domain_arg(
+        self,
+        dim_i: int,
+        dim: Dimension,
+        node_domain: past.Dict,
+    ) -> tuple[itir.Literal, itir.Literal]:
+        try:
+            if node_domain.keys_[dim_i].type.dim == dim:
+                domain_bounds = []
+                for domain_i, domain_input in enumerate(node_domain.values_[dim_i].elts):
+                    if isinstance(domain_input, past.Name):
+                        domain_bounds.append(itir.SymRef(id=domain_input.id))
+                    elif isinstance(domain_input, past.Constant):
+                        domain_bounds.append(
+                            itir.Literal(value=str(domain_input.value), type="int")
+                        )
```

Comment:
Ideally we would write this as something like
```suggestion
                    domain_bounds.append(self.visit(domain_input))
```
This already works for the case of `past.Constant` (it even does a better job because it has some checks, see `visit_Constant`), but not for `past.Name` because it would create a `Sym` instead of a `SymRef`. One solution could be to add a `as_ref` parameter to `visit_Name` (with default to False maybe) and then to a `self.visit(domain_input, as_ref=True)` here. Maybe ask for @tehrengruber 's feedback here.

Reply:
by doing that I get the following error for one of the tests:

TypeError: 'FunCall.args' must be <class 'functional.iterator.ir.Expr'> (got 'lower_i' which is a <class 'functional.iterator.ir.Sym'>

Reply:
I implemented a new approach, you can lmk if you want me to change it. Additionally, I added a test in the test_program for the error message in this section

Reply:
The current implementation looks much better, but I still prefer my original suggestion because we are doing extra steps that should be handled by the visitor. The error that you got suggests that you were lowering it as `Sym` instead of `SymRef`. Did you implement the `as_ref` pattern that I suggested?